Source/ASButtonNode.mm

```diff
@@ -131,6 +131,55 @@ - (void)setDisplaysAsynchronously:(BOOL)displaysAsynchronously
   [self.titleNode setDisplaysAsynchronously:displaysAsynchronously];
 }
 
+
+- (NSAttributedString *)__updateTitle:(NSAttributedString *)title withForegroundColor:(UIColor *)newColor
+{
+  if (title) {
+    NSMutableAttributedString *mutString = [[NSMutableAttributedString alloc] initWithAttributedString:title];
+    NSRange limit = NSMakeRange(0, _normalAttributedTitle.length);
+    NSRange effectiveRange;
+    UIColor *attributeValue = (UIColor *)[mutString attribute:NSForegroundColorAttributeName atIndex:limit.location effectiveRange:&effectiveRange];
+    if (attributeValue == nil) {
```

Comment:
Does this mean we are only setting the foreground color if one was set previously?

Reply:
We're only setting the tintColor as the titleColor if there was not one set previously

Reply:
I think this method name is confusing actually, will update